qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -300,6 +300,7 @@ def construct_expectation(self,
 
         observable_meas = self.expectation.convert(StateFn(operator, is_measurement=True))
         ansatz_circuit_op = CircuitStateFn(wave_function)
+        self._var_form_params = sorted(ansatz_circuit_op.parameters, key=lambda p: p.name)
```

Comment:
As the var_form_params are set sorted by the check_operator method, which is called ahead of this, in this same method, is this needed here, or just duplicates what was just done again? Neither the var form nor the operator should be None at this point I think whereby the sorted line in check_operator would be skipped.

Reply:
Sorry. I found out that this change is wrong. The root cause was the missing comparison of number of qubits in QAOA's _check_operator.